src/suggestions.js

```diff
@@ -0,0 +1,83 @@
+import {computeAccessibleName} from 'dom-accessibility-api'
+import {getRoles} from './role-helpers'
+import {getDefaultNormalizer} from './matches'
+
+const normalize = getDefaultNormalizer()
+
+function getLabelTextFor(element) {
+  let label
+
+  const allLabels = Array.from(document.querySelectorAll('label'))
+
+  label = allLabels.find(lbl => lbl.control === element)
+
+  if (!label) {
+    const ariaLabelledBy = element.getAttribute('aria-labelledby')
+
+    if (ariaLabelledBy) {
+      // we're using this notation because with the # selector we would have to escape special characters e.g. user.name
+      // see https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector#Escaping_special_characters
+      label = document.querySelector(`[id=${ariaLabelledBy}]`)
+    }
+  }
+
+  if (label) {
+    return label.textContent
+  }
+
+  return undefined
+}
+
+function makeSuggestion(queryName, content, name) {
+  return {
+    queryName,
+    toString() {
+      const options = name ? `, {name:/${name}/}` : ''
+      return `${queryName}("${content}"${options})`
+    },
+  }
+}
+
+export function getSuggestedQuery(element) {
+  const roles = getRoles(element)
+
+  const roleNames = Object.keys(roles)
+  const name = computeAccessibleName(element)
+  if (roleNames.length) {
+    const [role] = roleNames
+    return makeSuggestion('Role', role, name)
+  }
```

Comment:
No sense in computing the name if it's not used.

```suggestion
  if (roleNames.length) {
    const [role] = roleNames
    return makeSuggestion('Role', role, computeAccessibleName(element))
  }
```

Reply:
ah yeah good catch was playing w/ using that in the ByText but ended up not doing that...